File: ontrack-git/src/main/java/net/nemerosa/ontrack/git/GitRepository.java
package net.nemerosa.ontrack.git;

import lombok.Data;

import java.util.Objects;

/**
 * Coordinates for a remote repository.
 */
@Data
public class GitRepository {

    /**
     * Type of repository (source of information)
     */
    private final String type;

    /**
     * Name of the repository
     */
    private final String name;

    /**
     * Remote URL-ish for the repository.
     */
    private final String remote;

    /**
     * User. Blank when no authorisation is needed.
     */
    private final String user;

    /**
     * Password
     */
    private final String password;

    public String getId() {
        return (type + "_" + name + "_" + remote).replaceAll("[:\\.\\\\/@]", "_");
    }

}

File: ontrack-git/src/main/java/net/nemerosa/ontrack/git/GitRepository.kt
package net.nemerosa.ontrack.git

/**
 * Coordinates for a remote repository.
 *
 * @property type Type of repository (source of information)
 * @property name Name of the repository
 * @property remote Remote URL-ish for the repository.
 * @property authenticator Authentication mechanism to use for this repository
 */
class GitRepository(
    val type: String,
    val name: String,
    val remote: String,
    val authenticator: GitRepositoryAuthenticator? = null,
) {
    val id: String = (type + "_" + name + "_" + remote).replace("[:.\\\\/@]".toRegex(), "_")
}
